lib/idn.c

```diff
@@ -181,6 +242,8 @@ static CURLcode idn_decode(const char *input, char **output)
     result = CURLE_NOT_BUILT_IN;
 #elif defined(USE_WIN32_IDN)
   result = win32_idn_to_ascii(input, &decoded);
+#elif defined(HAVE_APPLE_IDN)
+  result = mac_idn_to_ascii(input, &decoded);
```

Comment:
```suggestion
  result = mac_idn_to_ascii(input, &decoded);
#else
  (void)input;
```
To avoid:
```
curl/lib/idn.c:235:40: error: unused parameter 'input' [-Werror=unused-parameter]
```

Reply:
This is no longer needed since we only set USE_IDN if we have the header, so the `#else` is not needed.